tools/licenses/lib/patterns.dart

```diff
@@ -4,281 +4,259 @@
 
 // ignore_for_file: use_raw_strings, prefer_interpolation_to_compose_strings
 
+import 'dart:core' hide RegExp;
+
+import 'regexp_debug.dart';
+
 // COMMON PATTERNS
 
+// _findLicenseBlocks in licenses.dart cares about the two captures here
+// it also handles the first line being slightly different ("/*" vs " *" for example)
 const String kIndent = r'^((?:[-;@#<!.\\"/* ]*(?:REM[-;@#<!.\\"/* ]*)?[-;@#<!.\"/*]+)?)( *)';
 
 final RegExp stripDecorations = RegExp(
-  r'^((?:(?:[-;@#<!.\\"/* ]*(?:REM[-;@#<!.\\"/* ]*)?[-;@#<!.\"/*]+)?)(?:(?: |\t)*))(.*?)[ */]*$',
+  r'^((?:(?:[-;@#<!.\\"/* ]*(?:REM[-;@#<!.\\"/* ]*)?[-;@#<!.\"/*]+)?) *)(.*?)[ */]*$',
   multiLine: true,
   caseSensitive: false
 );
 
-final RegExp newlinePattern = RegExp(r'\r\n?');
-
-final RegExp beginLicenseBlock = RegExp(
-  r'^([#;/* ]*) (?:\*\*\*\*\* BEGIN LICENSE BLOCK \*\*\*\*\*[ */]*'
-                r'|@APPLE_LICENSE_HEADER_START@)$'
-);
-
-final RegExp endLicenseBlock = RegExp(
-  r'^([#;/* ]*) (?:\*\*\*\*\* END LICENSE BLOCK \*\*\*\*\*[ */]*'
-                r'|@APPLE_LICENSE_HEADER_END@)$'
-);
+// _reformat patterns (see also irrelevantText below)
+final RegExp leadingDecorations = RegExp(r'^(?://|#|;|@|REM\b|/?\*|--| )');
+final RegExp fullDecorations = RegExp(r'^ *[-=]{2,}$');
+final RegExp trailingColon = RegExp(r':(?: |\*|/|\n|=|-)*$', expectNoMatch: true);
 
+final RegExp newlinePattern = RegExp(r'\r\n?');
 final RegExp nonSpace = RegExp('[^ ]');
-final RegExp trailingComma = RegExp(r',[ */]*$');
-final RegExp trailingColon = RegExp(r':(?: |\*|/|\n|=|-)*$');
-final RegExp copyrightMentionPattern = RegExp(r'©| \(c\) (?!{)|copy\s*right\b|copy\s*left', caseSensitive: false);
-final RegExp licenseMentionPattern = RegExp(r'license|warrant[iy]', caseSensitive: false);
+final RegExp copyrightSanityCheck = RegExp(r'©|copyright|\(c\)', caseSensitive: false);
+final RegExp copyrightMentionPattern = RegExp(r'©|copyright [0-9]|\(c\) [0-9]|copyright \(c\)', caseSensitive: false);
```

Comment:
Maybe add comments to distinguish these two from each other?

Reply:
added comments for these